### DISCUSSION TITLE
Import from Unreal Map in Carla

### DISCUSSION THREAD

**Discussion by prateekstark** [carla-simulator/carla#6104] [Q&A]
I have made a map in Unreal Engine, thus I have a .umap file. I want to use this map file for simulation in Carla but am not able to import and use it properly. Can anyone help me with this issue? Since the map was created in Unreal Engine itself, I do not have the OpenDrive file for that.

**Comment by kef-bassam** [comment]
You can generate the OpenDrive file using the Carla Python API: https://carla.readthedocs.io/en/latest/core_map/#navigation-in-carla
If that does not work, you may have to use the RoadRunner to generate a map and import into carla: https://carla.readthedocs.io/en/latest/tuto_M_generate_map/

**Comment by AloneMusk** [comment]
@prateekstark did this method work for you? I am trying to do the exact same thing and don't know where to start

**Reply by prateekstark** [reply]
No, it didn't but I ran the simulation using the umap file itself. The import issue was because of the incompatibility in versions of Carla Unreal Engine and the Map UE. Resolve that and you can then import in Carla UE.

**Reply by AloneMusk** [reply]
Ok so I am currently running the 9.15 version of Carla. This is the package file that's in a different Drive all together. I built carla from source since I needed the Unreal Editor and it's saved in another drive. Would this cause any issues? Also could you elaborate a bit on how exactly you solved the incompatibility issue?

**Reply by AloneMusk** [reply]
It would also be super helpful if you could just walk me through the steps of importing the map
